src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReflectionInvokeMapNode.cs

```diff
@@ -106,7 +106,7 @@ internal static void AddSignatureDependency(ref DependencyList dependencies, Nod
 
             TypeDesc canonType = type.ConvertToCanonForm(CanonicalFormKind.Specific);
             if (canonType.IsCanonicalSubtype(CanonicalFormKind.Any))
-                GenericTypesTemplateMap.GetTemplateTypeDependencies(ref dependencies, factory, type.ConvertToCanonForm(CanonicalFormKind.Specific));
+                GenericTypesTemplateMap.GetTemplateTypeDependencies(ref dependencies, factory, canonType);
             else
                 dependencies.Add(factory.MaximallyConstructableType(canonType), reason);
```

Comment:
Now that I'm actually looking at this, this one is also a bit weird (type == canonType in this branch so they're equivalent, but using type makes more sense).

```suggestion
                dependencies.Add(factory.MaximallyConstructableType(type), reason);
```